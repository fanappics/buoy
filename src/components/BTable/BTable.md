<docs>
Sample documentation for BTable

```js
Add code examples here!
```
</docs>
