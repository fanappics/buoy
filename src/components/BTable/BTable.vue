<template>
  <div class="b-table">
    <table class="b-table-el">
      <tr class="b-table-row">
        <th 
          v-for="header in headers"
          class="b-table-header"
        >
          <button
          type="button"
          class="header-button"
          @click="sortColumn(header.key)"
          :disabled="!header.filter"
          >
            {{ header.name }}<img v-if="header.name && header.filter" class="icon" :src="unsorted" />
          </button>
        </th>
      </tr>
      <tr
        v-for="row in rows"
        class="b-table-row"
      >
        <td
          v-for="cell in row"
          class="b-table-cell"
          v-html='cell'
        />
        </td>
      </tr>
    </table>
    <div class="pagination">
      <div class="pagination-buttons">
        <button
          type="button" 
          class="pagination-button"
          @click="paginate(currentPage-1)"
          :disabled="currentPage <= 0"
        >
          <i class='pagination-icon ion-arrow-left-c' aria-hidden='true'></i>
        </button>
        <button
          type="button"
          class="pagination-button"
          @click="paginate(currentPage+1)"
        >
          <i class='pagination-icon ion-arrow-right-c' aria-hidden='true'></i>
        </button>
      </div>
    </div>
  </div>
</template>

<script>

export default {

  name: 'b-table',
  props: {
    id: {
      type: String,
      required: true
    },
    label: {
      type: String,
      required: true
    },
    headers: {
      type: Array,
      required: true
    },
    tableData: {
      type: Array,
      required: true
    },
    pageNumber: {
      type: Number,
      required: false
    },
    rowsPerPage: {
      type: Number,
      default: 10
    },
    totalResults: {
      type: Number,
    }
  },
  data () {
    return {
      rows: this.buildRows(this.tableData, this.headers),
      unsorted: require('../../../static/unSortedArrows.svg'),
      currentPage: this.pageNumber ? this.pageNumber : 0
    }
  },
  methods: {
    buildRows: function(rowData, headers) {
      let rows = []
      rowData.forEach( function(data) {
        let row = []
        headers.forEach( function(header) {
          if (header.render) {
            row.push(header.render(data[header.key]))
          } else {
            row.push(data[header.key])
          }
        })
        rows.push(row)
      })
      return rows
    },
    sortColumn: function(sortBy) {
      this.$emit("sort", sortBy)
    },
    paginate: function(page) {
      this.$emit("paginate", page)
      this.currentPage = page
      console.log(this.currentPage)
    }
  }

}
</script>

<style scoped>

  .b-table-el {
    width: 100%;
    border-collapse: collapse;
  }
  
  .b-table-row {
    height: 4.5rem;
  }

  td {
    padding-left: .825rem
  }
  
  tr:nth-child(even) {
    background-color: #f5f6f8;
  }

  th {
    border-bottom: solid;
    border-width: thin;
    border-color: #c4c3c4;
    text-align: left;
  }

  button {
    border: none;
    font-weight: 600;
  }

  button:focus {
    outline: 0;
    border: none;
  }

  button.header-button:disabled {
    color: #000000;
    opacity: 100;
    cursor: default;
  }

  img.icon {
    position: absolute;
    margin-left: .725rem;
  }

  .pagination {
    border-style: solid none solid none;
    border-width: thin;
    width: 100%;
    text-align: right;
    height: 2.5rem;
    border-color: #c4c3c4;
  }

  .pagination-icon {
    color: #4D90FE;
    font-size: 22px;
  }

  .pagination-buttons{
    padding-right: .5rem;
  }
  .pagination-button{
    padding: 0 .5rem;
    margin-top: .5rem
  }

</style>
