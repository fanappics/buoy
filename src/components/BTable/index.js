import BTable from './BTable.vue'

/* istanbul ignore next */
BTable.install = function install (Vue) {
  Vue.component(BTable.name, BTable)
}

export default BTable
