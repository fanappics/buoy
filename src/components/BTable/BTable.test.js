import { mount } from 'vue-test-utils'
import BTable from './BTable'

describe('BTable', () => {
  const label = 'test label'
  const testId = 'test id'
  const testHeaders = [
    { name: 'testColumn1', key: 'testkey1' }, { name: 'testColumn2', key: 'testkey2' }
  ]
  const tableData = [
    { header_one: 'column 1', header_two: 'column 2', header_three: 'column 3', header_four: 'column 4', header_five: 'column 5', header_six: 'column 6', name: 'example row 1' }
  ]
  const totalResults = 100

  it('BTable renders correctly', () => {
    const wrapper = mount(BTable, {
      propsData: {
        id: testId,
        label: label,
        headers: testHeaders,
        tableData: tableData,
        totalResults: totalResults
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
