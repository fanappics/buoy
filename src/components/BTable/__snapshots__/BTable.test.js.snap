// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BTable BTable renders correctly 1`] = `
<table class="b-table">
    <tr class="b-table-row">
        <th class="b-table-header">
            testColumn1
        </th>
        <th class="b-table-header">
            testColumn2
        </th>
    </tr>
    <tr class="b-table-row">
        <td class="b-table-cell"></td>
        <td class="b-table-cell"></td>
    </tr>
</table>
`;
